pkg/sources/redisstreams/redisstream.go

```diff
@@ -228,6 +229,93 @@ func newRedisClient(sourceSpec *dfv1.RedisStreamsSource) (*redisclient.RedisClie
 	return redisclient.NewRedisClient(opts), nil
 }
 
+// incoming message should have multiple key/value pairs: produce a message with no Key, with value set to JSON serialization of the pairs
+func produceUnkeyedJSONMsg(inMsg redis.XMessage) (*isb.ReadMessage, error) {
+	var readOffset = inMsg.ID
+
+	// JSON serialize our key/value pairs
+	// need to first convert map[string]interface to map[string]string
+	kvMap := make(map[string]string)
+	for k, v := range inMsg.Values {
+		strValue, castOk := v.(string)
+		if !castOk {
+			return nil, fmt.Errorf("couldn't cast RedisStream interface type %v to string: inMsg=%+v", v, inMsg)
+		}
+		kvMap[k] = strValue
+	}
+	jsonSerialized, err := json.Marshal(&kvMap)
+	if err != nil {
+		return nil, fmt.Errorf("failed to json serialize RedisStream values: %v; inMsg=%+v", err, inMsg)
+	}
+
+	msgTime, err := msgIdToTime(inMsg.ID)
+	if err != nil {
+		return nil, err
+	}
+	isbMsg := isb.Message{
+		Header: isb.Header{
+			MessageInfo: isb.MessageInfo{EventTime: msgTime},
+			ID:          readOffset,
+		},
+		Body: isb.Body{Payload: []byte(jsonSerialized)},
+	}
+
+	return &isb.ReadMessage{
+		ReadOffset: isb.SimpleStringOffset(func() string { return readOffset }),
+		Message:    isbMsg,
+	}, nil
+}
+
```

Comment:
@vigith How do you feel about this being a json map? (since you specifically mentioned a "json serialized array of key, value pairs") This should work just as well, right?